File: cpp_prac/kakaosecretmp.cpp
#include <string>
#include <vector>
#include<iostream>
using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    int temp[n];
    for(int i=0; i<n; i++) temp[i]=arr1[i] | arr2[i];
    string line;
    for(int i=0; i<n; i++)
    {
        for(int j=0; j<n; j++)
        {
            if(temp[i]%2==1) line.insert(0,1,'#');
            else line.insert(0,1,' ');
            temp[i]/=2;
        }
        answer.push_back(line);
        line.clear();
    }
    return answer;
}

File: java_prac/kakaosecretmp.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        for (int i = 0; i < n; ++i) {
            int num = arr1[i] | arr2[i];
            char pres[] = new char[n];
            int index = n - 1;
            while (num != 0) {
                if (num % 2 == 1)
                    pres[index--] = '#';
                else
                    pres[index--] = ' ';
                num /= 2;
            }
            for (int k = index; k >= 0; --k) {
                pres[k] = ' ';
            }
            answer[i] = String.valueOf(pres);
        }
        return answer;
    }
}